src/codec.js

```diff
@@ -69,6 +69,39 @@ Int.prototype.valueOf = function() {
 
 codec.Int = Int;
 
+var TYPE = Symbol();
+
+var Struct = {
```

Comment:
@stephenplusplus I know this isn't what we initially discussed, but I ran into some weirdness when subclassing Arrays, so I decided not to. LMK if you'd prefer I organize/structure this differently, I know singletons aren't really something we do.

Reply:
I can go either way between this being the way it is now or if we went with a Class system. You can return the array from a class constructor, so it should be the same. `createFromJSON` and `isStruct` could be static methods. The reason I prefer that is for consistency with our typical usage, and the reason I don't is because you already didn't do it that way. So again, I can go either way :)